scripts/linters/js_ts_linter.py

```diff
@@ -0,0 +1,930 @@
+                        angular_constant_name = (
+                            angular_constant_node.expression.left.property.name)
+                        angular_constants_list.append(angular_constant_name)
+
+                    angular_constants_set = set(angular_constants_list)
+                    if len(angular_constants_set) != len(
+                            angular_constants_list):
+                        failed = True
+                        python_utils.PRINT(
+                            '%s --> Duplicate constant declaration found.'
+                            % filepath)
+                    if corresponding_angularjs_filepath in (
+                            angularjs_source_filepaths_to_constants_dict):
+                        angular_minus_angularjs_constants = (
+                            angular_constants_set.difference(
+                                angularjs_source_filepaths_to_constants_dict[
+                                    corresponding_angularjs_filepath]))
+                        for constant in angular_minus_angularjs_constants:
+                            failed = True
+                            python_utils.PRINT(
+                                '%s --> The constant %s is not declared '
+                                'in the corresponding angularjs '
+                                'constants file.' % (filepath, constant))
+
+            if failed:
+                summary_message = (
+                    '%s  Constants declaration check failed, '
+                    'see messages above for constants with errors.' % (
+                        _MESSAGE_TYPE_FAILED))
+            else:
+                summary_message = '%s  Constants declaration check passed' % (
+                    _MESSAGE_TYPE_SUCCESS)
+            summary_messages.append(summary_message)
+            python_utils.PRINT(summary_message)
+
+        return summary_messages
+
+    def perform_all_lint_checks(self):
+        """Perform all the lint checks and returns the messages returned by all
+        the checks.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+        all_messages = []
+
+        if not self.all_filepaths:
+            python_utils.PRINT('')
+            python_utils.PRINT(
+                'There are no JavaScript or Typescript files to lint.')
+            return []
+
+        self.parsed_js_and_ts_files = self._validate_and_parse_js_and_ts_files()
+        self.parsed_expressions_in_files = (
+            self._get_expressions_from_parsed_script())
+
+        all_messages.extend(self._check_extra_js_files())
+
+        all_messages.extend(self._check_js_and_ts_component_name_and_count())
+
+        all_messages.extend(self._check_directive_scope())
+
+        all_messages.extend(self._check_sorted_dependencies())
+
+        all_messages.extend(
+            self._match_line_breaks_in_controller_dependencies())
```

Comment:
Sorry, I think I might have asked you to use `.extend` before. Can you change it to `all_messages += ...`? This is actually for a minor performance boost and it's easier to read.

Reply:
Done.